test/detection/test_line_counter.py

```diff
@@ -70,3 +73,320 @@ def test_calculate_region_of_interest_limits(
+            ],
+            [False, True],
+            [False, False],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [-100, 50, -80, 70],
+                [-10, 50, 20, 70],
+                [100, 50, 120, 70],
+            ],
+            [False, False, True],
+            [False, False, False],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(100, 100),
+            ),
+            [
+                [50, 45, 70, 30],
+                [40, 50, 50, 40],
+                [0, 50, 10, 40],
+            ],
+            [False, False, False],
+            [False, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(100, 0),
+            ),
+            [
+                [50, -45, 70, -30],
+                [40, 50, 50, 40],
+            ],
+            [False, False],
+            [False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, -100),
+            ),
+            [
+                [100, -50, 120, -70],
+                [-100, -50, -80, -70],
+            ],
+            [False, True],
+            [False, False],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(50, 100),
+            ),
+            [
+                [50, 50, 70, 30],
+                [40, 50, 50, 40],
+                [0, 50, 10, 40],
+            ],
+            [False, False, False],
+            [False, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(0, 100),
+            ),
+            [
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+                [100, 50, 120, 70],
+                [-100, 50, -80, 70],
+            ],
+            [False, False, True, False, True, False, True, False],
+            [False, True, False, True, False, True, False, True],
+        ),
+        (
+            Vector(
+                Point(0, 0),
+                Point(-100, 0),
+            ),
+            [
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+                [-50, 70, -40, 50],
+                [-50, -70, -40, -50],
+            ],
+            [False, False, True, False, True, False, True, False],
+            [False, True, False, True, False, True, False, True],
+        ),
+    ],
+)
+def test_line_zone_single_detection(
+    vector, bbox_sequence, expected_count_in: List[bool], expected_count_out: List[bool]
```

Comment:
Once again, please rename `expected_count_in` and `expected_count_out` to `expected_crossed_in` and `expected_crossed_out` respectively.

Reply:
Provide type hints for all arguments.